src/IoTeX/Staking.cpp

```diff
@@ -0,0 +1,211 @@
+// Copyright © 2017-2020 Trust Wallet.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+#include "Staking.h"
+
+using namespace TW;
+
+namespace TW::IoTeX {
+
+/// Pack uint64 to uint256 in big-endian order
+///
+Data packUint64ToUint256(uint64_t value)
+{
+    Data result(32);
+
+    for (auto it = result.rbegin(); it < result.rend(); it++) {
+        *it = (byte)value;
+        if ((value = value >> 8) == 0)
+            break;
+    }
+    return result;
+}
+
+/// Pack byte array to end of calldata
+///
+/// \param dataLen length of the last input _data
+/// \param _data payload bytes that user wants to write
+///
+Data packBytesToEnd(uint64_t offset, uint64_t dataLen, const byte *_data)
+{
+    Data result;
+
+    // pack offset
+    auto off = packUint64ToUint256(offset);
+    append(result, off);
+
+    // pack data length
+    auto length = packUint64ToUint256(dataLen);
+    for (int i = 0; i < 32; i++) {
+        append(result, length[i]);
+    }
+    // pack _data
+    for (int i = 0; i < dataLen; i++) {
+        append(result, _data[i]);
+    }
+    // pad to 32-aligned
+    if (dataLen & 31) {
+        int padding = 32 - (dataLen & 31);
+        for (int i = 0; i < padding; i++) {
+            append(result, 0);
+        }
+    }
+    return result;
+}
+
+/// Pack the calldata for 'createPygg'
+///
+/// \param _canName delegate name, must be 12 bytes long
+/// \param _stakeDuration days to stake, must be multiple of 7
+/// \param _nonDecay if true, the stake time keeps at _stakeDuration days
+/// \param dataLen length of the last input _data
+/// \param _data payload bytes that user wants to write
+///
+Data Stake(
```

Comment:
- Some method names start with lowercase, some with uppercase -- is there any logic to it?  We use camelCase.

Reply:
uppercase is API to be called by outside caller
lowercase is utility func used inside by uppercase API func

let me know if you want to use camelCase for all

Reply:
It's fine this way.